File: DFS of graph/solve.cpp
class Solution {
  public:
    vector<int> ans;
    
    // Function to return a list containing the DFS traversal of the graph.
    vector<int> dfsOfGraph(int V, vector<int> adj[]) {
        unordered_map<int, bool> visited;
        
        DFS(adj, visited, 0);
        
        return ans;
    }
    
    void DFS(vector<int> adj[], unordered_map<int, bool>& visited, int curr) {
        visited[curr] = true;
        ans.push_back(curr);
        
        for(auto i : adj[curr]) {
            if(!visited[i]) {
                DFS(adj, visited, i);
            }
        }
    }
};

File: DFS of graph/solve.java
class Solution {
    // Function to return a list containing the DFS traversal of the graph.
    public ArrayList<Integer> dfsOfGraph(int V, ArrayList<ArrayList<Integer>> adj) {
        boolean [] vis = new boolean[V];
        ArrayList<Integer> ans = new ArrayList<>();
        for(int i=0; i<V; i++){
            if(vis[i] == false){
                storingNodes(ans, vis, i, adj);
            }
        }
        return ans;
    }
    
    public static void storingNodes(ArrayList<Integer> ans, boolean [] vis, int i, ArrayList<ArrayList<Integer>> adj){
        ans.add(i);
        vis[i] = true;
        for(int neighbours : adj.get(i)){
            if(vis[neighbours] == false){
                storingNodes(ans, vis, neighbour, adj);
            }
        }
    }
}
